From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: zfcp: no port recovery after storage side error inject
References: bnc#487755,LTC#52402

Symptom:     The remote port remains in error state even if connection
             is re-established.
Problem:     A wrong precondition check was performed on the port status
             leading to a cancellation of the port reopen.
Solution:    Remove the pre-req check because it's not required and better
             handled within the ERP.

Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/scsi/zfcp_fc.c |    5 -----
 1 file changed, 5 deletions(-)

Index: linux-sles11/drivers/s390/scsi/zfcp_fc.c
===================================================================
--- linux-sles11.orig/drivers/s390/scsi/zfcp_fc.c
+++ linux-sles11/drivers/s390/scsi/zfcp_fc.c
@@ -430,11 +430,6 @@ void zfcp_fc_link_test_work(struct work_
 		container_of(work, struct zfcp_port, test_link_work);
 	int retval;
 
-	if (!(atomic_read(&port->status) & ZFCP_STATUS_COMMON_UNBLOCKED)) {
-		zfcp_port_put(port);
-		return; /* port erp is running and will update rport status */
-	}
-
 	zfcp_port_get(port);
 	port->rport_task = RPORT_DEL;
 	zfcp_scsi_rport_work(&port->rport_work);
